src/state/actions/resourceActions.ts

```diff
@@ -0,0 +1,58 @@
+import { IStructureDefinition } from "@ahryman40k/ts-fhir-types/lib/R4";
+
+export const SELECT_RESOURCE = 'SELECT_RESOURCE';
+export const GET_DATA_START = 'GET_DATA_START';
+export const GET_DATA_SUCCESS = 'GET_DATA_SUCCESS';
+export const GET_DATA_FAILURE = 'GET_DATA_FAILURE';
+
+
+export type SelectResourceAction = {
+    type: typeof SELECT_RESOURCE,
+    id: string
+}
+
+export type GetDataStartAction = {
+    type: typeof GET_DATA_START,
+    payload: boolean
+}
+
+export type GetDataSuccessAction = {
+    type: typeof GET_DATA_SUCCESS,
+    payload: IStructureDefinition[]
+}
+
+export type GetDataFailureAction = {
+    type: typeof GET_DATA_FAILURE,
+    payload: string
+}
+
+export const selectResource = (id: string): SelectResourceAction => {
+    return {
+        type: SELECT_RESOURCE,
+        id
+    }
+}
+
+export const getDataStart = (loading: boolean): GetDataStartAction => {
```

Comment:
'loading' argument is useless as we already know the app is loading whenever this action is dispatched